# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
# Commit: ed2e505b5873
# Full Hash: ed2e505b587309427804b91ddace88dd22624711
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2021-11-01 18:17:33
# Regressor Bug: 1654112
# File Overlap Count: 1
# Description:
#   Bug 1654112 - In VideoConduit::SendVideoFrame protect against racy setters. r=bwc
#   
#   With VideoConduit setters being called on a worker thread rather than main, we
#   no longer have a guarantee that the first SendVideoFrame comes in after ssrcs
#   or the send codec config are set.
# ==============================================================================

diff -r df9a58057245 -r ed2e505b5873 dom/media/webrtc/libwebrtcglue/VideoConduit.cpp
--- a/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Sun Apr 18 21:32:34 2021 +0200
+++ b/dom/media/webrtc/libwebrtcglue/VideoConduit.cpp	Wed Apr 21 01:13:43 2021 +0200
@@ -1234,6 +1234,16 @@
   int adaptedHeight;
   {
     MutexAutoLock lock(mMutex);
+    if (mSendStreamConfig.rtp.ssrcs.empty()) {
+      CSFLogVerbose(LOGTAG, "WebrtcVideoConduit %p %s No SSRC set", this,
+                    __FUNCTION__);
+      return kMediaConduitNoError;
+    }
+    if (!mCurSendCodecConfig) {
+      CSFLogVerbose(LOGTAG, "WebrtcVideoConduit %p %s No send codec set", this,
+                    __FUNCTION__);
+      return kMediaConduitNoError;
+    }
     CSFLogVerbose(LOGTAG, "WebrtcVideoConduit %p %s (send SSRC %u (0x%x))",
                   this, __FUNCTION__, mSendStreamConfig.rtp.ssrcs.front(),
                   mSendStreamConfig.rtp.ssrcs.front());
